packages/frontend-2/components/settings/workspaces/General/DeleteDialog.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+  <LayoutDialog
+    v-model:open="isOpen"
+    title="Delete workspace"
+    max-width="sm"
+    :buttons="dialogButtons"
+  >
+    <p>
+      Are you sure you want to
+      <span class="font-medium">permanently delete</span>
+      the selected workspace?
+    </p>
+    <div
+      class="rounded border bg-foundation-2 border-outline-3 text-body-2xs text-foreground font-medium py-3 px-4 my-4"
+    >
+      {{ workspace.name }}
+    </div>
+    <p>
+      This action
+      <span class="font-medium">cannot</span>
+      be undone.
+    </p>
+  </LayoutDialog>
+</template>
+<script setup lang="ts">
+import { graphql } from '~~/lib/common/generated/gql'
+import type {
+  SettingsWorkspaceGeneralDeleteDialog_WorkspaceFragment,
+  WorkspaceCollection
+} from '~/lib/common/generated/gql/graphql'
+import type { LayoutDialogButton } from '@speckle/ui-components'
+import { useMutation, useApolloClient } from '@vue/apollo-composable'
+import { deleteWorkspaceMutation } from '~/lib/settings/graphql/mutations'
+import {
+  convertThrowIntoFetchResult,
+  getFirstErrorMessage,
+  getCacheId,
+  modifyObjectFields
+} from '~~/lib/common/helpers/graphql'
+import { ToastNotificationType, useGlobalToast } from '~~/lib/common/composables/toast'
+import { useActiveUser } from '~~/lib/auth/composables/activeUser'
+
+graphql(`
+  fragment SettingsWorkspaceGeneralDeleteDialog_Workspace on Workspace {
+    id
+    name
+  }
+`)
+
+const props = defineProps<{
+  workspace: SettingsWorkspaceGeneralDeleteDialog_WorkspaceFragment
+}>()
+
+const isOpen = defineModel<boolean>('open', { required: true })
+
+const { mutate: deleteWorkspace } = useMutation(deleteWorkspaceMutation)
+const { triggerNotification } = useGlobalToast()
+const { activeUser } = useActiveUser()
+const apollo = useApolloClient().client
+
+const onDelete = async () => {
+  isOpen.value = false
+
+  const cache = apollo.cache
+  const result = await deleteWorkspace({
+    workspaceId: props.workspace.id
+  }).catch(convertThrowIntoFetchResult)
+
+  if (result?.data) {
+    if (activeUser.value) {
+      cache.evict({
+        id: getCacheId('Workspace', props.workspace.id)
+      })
+
+      modifyObjectFields<{ workspaces: WorkspaceCollection }, WorkspaceCollection>(
+        cache,
+        activeUser.value.id,
+        (fieldName, _variables, value) => {
+          return {
+            ...value,
```

Comment:
There's a reason all of the fields of `value` are optional - you don't know which if any of those fields are actually going to be requested and in the cache. If `totalCount` is undefined, meaning - not requested, you can't just default to 0. Just because we didn't ask for a totalCount from the API (the list may actually have a 100 workspaces), doesn't mean that we should now show it has 1 on the frontend.

The correct way to handle this is to only update totalCount if it exists in the cache. If it doesn't - leave it as is.

```
const newTotalCount = isUndefined(value?.totalCount) ? undefined : Math.max(0, (value?.totalCount || 0) - 1)
return {
  ...value,
  ...(isUndefined(newTotalCount) ? {} : {totalCount: newTotalCount})
}
```

Reply:
Thanks for the explanation, not super familiar with best practices regarding this so it's helpful :) 
So I've updated it like this:

```      modifyObjectFields<UserWorkspacesArgs, User['workspaces']>(
        cache,
        activeUser.value.id,
        (_fieldName, variables, value, { DELETE }) => {
          if (variables?.filter?.search?.length) return DELETE

          const newTotalCount = isUndefined(value?.totalCount)
            ? undefined
            : Math.max(0, (value?.totalCount || 0) - 1)

          return {
            ...value,
            ...(isUndefined(newTotalCount) ? {} : { totalCount: newTotalCount })
          }
        },
        { fieldNameWhitelist: ['workspaces'] }
      )```
      
      So basically we first check if it's part of a search filter (which it can't be now, but I guess makes it more future proof), and then update the totalCount if it's set in the cache, is this the correct approach?